detection_rules/rule.py

```diff
@@ -363,6 +364,7 @@ class RelatedIntegrations:
     references: Optional[List[str]]
     related_integrations: Optional[List[RelatedIntegrations]] = field(metadata=dict(metadata=dict(min_compat="8.3")))
     required_fields: Optional[List[RequiredFields]] = field(metadata=dict(metadata=dict(min_compat="8.3")))
+    revision: Optional[definitions.PositiveInteger] = field(metadata=dict(metadata=dict(min_compat="8.8")))
```

Comment:
just leaving this here for future consideration (if we add any more of these fields where they are valid for anyone but Elastic). We could abstract this using more field metadata as a generic `non_elastic_field` (or whatever)